reparse.go

```diff
@@ -126,3 +138,83 @@ func EncodeReparsePoint(rp *ReparsePoint) []byte {
 	binary.Write(&b, binary.LittleEndian, target16)
 	return b.Bytes()
 }
+
+// ReparseDataBuffer corresponds to REPARSE_DATA_BUFFER struct defined here:
+// https://docs.microsoft.com/en-us/windows-hardware/drivers/ddi/ntifs/ns-ntifs-_reparse_data_buffer.
+// ReparesPoint struct (and corresponding Encode/Decode functions) provided above only
+// handle symlink or mountpoint type of reparse points. To work with other reparse points
+// use this struct.  When calling SetReparsePoint, if it is a symlink or mountpoint
+// reparse point pass in the buffer generated by EncodeReparsePointData. If it is some
+// other kind of reparse point pass in the buffer generated by
+// ReparseDataBuffer.Encode. (same goes for GetReparsePoints)
+type ReparseDataBuffer struct {
+	ReparseTag        uint32
+	ReparseDataLength uint16
+	Reserved          uint16
+	// a generic data buffer
+	DataBuffer []byte
+}
+
+// Encode encodes the current ReparseDataBuffer struct that can be passed
+// to the SetReparsePoint.
+func (r *ReparseDataBuffer) Encode() []byte {
+	var buf bytes.Buffer
+	binary.Write(&buf, binary.LittleEndian, r.ReparseTag)
+	binary.Write(&buf, binary.LittleEndian, r.ReparseDataLength)
+	binary.Write(&buf, binary.LittleEndian, r.Reserved)
+	buf.Write(r.DataBuffer)
+	return buf.Bytes()
+}
+
+// Decode decodes the byte array (returned by GetReparsePoint) into
+// the current ReparseDataBuffer struct.
+func (r *ReparseDataBuffer) Decode(data []byte) error {
+	rdr := bytes.NewReader(data)
+	err := binary.Read(rdr, binary.LittleEndian, r)
+	if err != nil {
+		return errors.Wrap(err, "failed to decode reparse data buffer")
```

Comment:
There was actually a recent change that aimed at getting rid of the github.com/pkg/errors package in this repo 😆 https://github.com/microsoft/go-winio/pull/227/files. So best to swap to the stdlib form here and for the other wraps throughout

Reply:
Fixed.